### PR TITLE
Initial commit on skydns + dashboard

### PR DESCRIPTION
**Pull request by tayzlor**
Still on the TODO - 
- [x] We need a solution for calling kubectl to provision the addons on the first master. This probably involves setting up kubectl there properly and integrating it with some form of wrapper. So blocking this on https://github.com/Capgemini/apollo-kubernetes/issues/2

### CODE REVIEW

File: roles/addons/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+# defaults file for addons
+
+# determines if we are on the first master instance.
+is_first_master: "{% if inventory_hostname == groups['masters'][0] %}true{% else %}false{% endif %}"
```
Comment (enxebre) [inline roles/addons/defaults/main.yml]: Any reason you using this instead of ansible "run_once"?

Comment (tayzlor) [inline roles/addons/defaults/main.yml]: nope, just an oversight by me - probably could be run_once i think

Comment (enxebre) [inline roles/addons/defaults/main.yml]: cool, we'll create pr for making kubectl available then we can discuss how to merge/integrate

**Comment by enxebre** [comment]
First approach for kubectl here https://github.com/Capgemini/kubeform/pull/50/files

**Comment by tayzlor** [comment]
merged in https://github.com/Capgemini/kubeform/commit/636be837cb0a84b631b57e0504dc41833531de0a i'll rebase this

**Comment by tayzlor** [comment]
Updated code here. It works on first shot provision but if you re-run you get - 

```
fatal: [kube-master-0]: FAILED! => {"changed": true, "cmd": ["/opt/bin/kubectl", "create", "-f", "/etc/kubernetes/manifests/skydns-rc.yaml"], "delta": "0:00:00.325062", "end": "2016-03-22 14:16:16.705597", "failed": true, "rc": 1, "start": "2016-03-22 14:16:16.380535", "stderr": "Error from server: error when creating \"/etc/kubernetes/manifests/skydns-rc.yaml\": replicationControllers \"kube-dns-v9\" already exists", "stdout": "", "stdout_lines": [], "warnings": []}```

thats because the registered variable isn't held across ansible runs (only in memory for the duration of the run). I think we could potentially get this PR in though and improve on it

**Comment by tayzlor** [comment]
As a way forward I see 3 options - 

- drop a file on the remote system so that we can tell if a svc/rc/etc has already been provisioned via kubectl
- be more verbose in our playbook and call ```kubectl get -f filename.yaml``` grab the output parse it and see if its been created, based on that call an apply or a create
- create an ansible module wrapper around kubectl and handle the checking at the module level (see https://github.com/Capgemini/kubeform/issues/2)

My preference would be the 3rd option.

**Comment by wallies** [comment]
yeh i see option 2 or 3 being a lot nicer. For now probably option 2, then create an issue to do option 3 as an improvement when we add more add-ons and services

**Comment by enxebre** [comment]
This looks good enough to me to go, lets create follow ups to improve.

**Pull request merged**